### PR TITLE
enable SSL for curl in bazel build environment

### PR DESCRIPTION
**Pull request by SF-Zhou**

### CODE REVIEW

File: bazel/curl.BUILD [hunk 1]
```diff
@@ -55,5 +57,9 @@ cc_library(
             "-lpthread",
         ],
     }),
+    deps = [
+        "@boringssl//:crypto",
+        "@boringssl//:ssl",
+    ],
```
Comment (lalitb) [inline bazel/curl.BUILD:63]: The CMake build has a dependency on OpenSSL. Can we have the same dependency to maintain consistency?

File: bazel/curl.BUILD [hunk 2]
```diff
@@ -34,6 +34,8 @@ cc_library(
     ]),
     copts = CURL_COPTS + [
         "-DOS=\"os\"",
+        "-DCURL_WITH_MULTI_SSL",
```
Comment (ThomsonTan) [inline bazel/curl.BUILD:37]: What is the option `CURL_WITH_MULTI_SSL` for?

File: bazel/curl.BUILD (hunk 1 above)
Comment (owent) [reply bazel/curl.BUILD:62]: Agree, but gRPC also use boringssl, maybe we can use alias to switch the ssl library. (https://bazel.build/docs/configurable-attributes?#faq-select-bind)

**Comment by marcalff** [comment]
The discussion on this code review is inconclusive, as there is no agreement on which SSL library to use to build CURL, and the topic needs further investigation.

Created issue #2489 to document the bug found, and closing this PR.

**Pull request closed without merging**